frontend/lib/wallets/MetaMask.ts

```diff
@@ -41,7 +55,19 @@ export default class MetaMask extends AbstractWallet {
     }
 
     getProvider(): ethers.providers.JsonRpcProvider {
-        return new ethers.providers.Web3Provider(window.ethereum, 'any');
+        if (!MetaMask.provider) {
+            MetaMask.provider = window.ethereum;
```

Comment:
this would already assign a provider, which might not be a metamask provider (since it can be coinbase, as we learned). Can this be done more safe? Ie, is there a way to check if it's a metamask by the metamask API itself?